tests/src/JIT/Regression/JitBlue/GitHub_7508/Vector3Test.cs

```diff
@@ -0,0 +1,136 @@
+using System.Runtime.CompilerServices;
+using System.Numerics;
+
+namespace Test01
+{
+    public struct SimpleVector3
+    {
+        [MethodImpl( MethodImplOptions.AggressiveInlining )]
+        public SimpleVector3( float x, float y, float z )
+        {
+            this.x = x;
+            this.y = y;
+            this.z = z;
+        }
+
+        [MethodImpl( MethodImplOptions.AggressiveInlining )]
+        public static SimpleVector3 operator +( SimpleVector3 a, SimpleVector3 b )
+            => new SimpleVector3( a.x + b.x, a.y + b.y, a.z + b.z );
+
+        public float X
+        {
+            get { return x; }
+            set { x = value; }
+        }
+
+        public float Y
+        {
+            get { return y; }
+            set { y = value; }
+        }
+
+        public float Z
+        {
+            get { return z; }
+            set { z = value; }
+        }
+
+        float x, y, z;
+    }
+
+    public struct WrappedVector3
+    {
+        [MethodImpl( MethodImplOptions.AggressiveInlining )]
+        public WrappedVector3( float x, float y, float z )
+        {
+            v = new System.Numerics.Vector3( x, y, z );
+        }
+
+        [MethodImpl( MethodImplOptions.AggressiveInlining )]
+        WrappedVector3( System.Numerics.Vector3 v )
+        {
+            this.v = v;
+        }
+
+        [MethodImpl( MethodImplOptions.AggressiveInlining )]
+        public static WrappedVector3 operator +( WrappedVector3 a, WrappedVector3 b )
+            => new WrappedVector3( a.v + b.v );
+
+        public float X
+        {
+            get { return v.X; }
+            set { v.X = value; }
+        }
+
+        public float Y
+        {
+            get { return v.Y; }
+            set { v.Y = value; }
+        }
+                
+        public float Z
+        {
+            get { return v.Z; }
+            set { v.Z = value; }
+        }
+
+        Vector3 v;
+    }
+
+    public class Program
+    {
+        static Random random = new Random( 12345 );
+        [MethodImpl( MethodImplOptions.NoInlining )]
+        static SimpleVector3 RandomSimpleVector3()
+            => new SimpleVector3( (float)random.NextDouble(), (float)random.NextDouble(), (float)random.NextDouble() );
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static WrappedVector3 RandomWrappedVector3()
+            => new WrappedVector3( (float)random.NextDouble(), (float)random.NextDouble(), (float)random.NextDouble() );
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static Vector3 RandomVector3()
+            => new Vector3( (float)random.NextDouble(), (float)random.NextDouble(), (float)random.NextDouble() );
+
+        public static void TestSimple()
+        {
+            var simpleA = RandomSimpleVector3();
+            var simpleB = RandomSimpleVector3();
+            var simpleC = simpleA + simpleB;
+            Console.WriteLine("Simple Vector3: {0},{1},{2}", simpleC.X, simpleC.Y, simpleC.Z);
+        }
```

Comment:
All the output seems pretty useless to an observer; I suppose it's only use is to prevent optimization.